packages/hub/card-services.ts

```diff
@@ -0,0 +1,233 @@
+import Session from '@cardstack/plugin-utils/session';
+import { declareInjections } from '@cardstack/di';
+import { todo } from '@cardstack/plugin-utils/todo-any';
+import { SingleResourceDoc, ResourceObject, ResourceIdentifierObject, CollectionResourceDoc } from 'jsonapi-typescript';
+import { get, uniqBy } from 'lodash';
+
+const cardIdDelim = '::';
+const cardBrowserAssetFields = [
+  'isolated-template',
+  'isolated-js',
+  'isolated-css',
+  'embedded-template',
+  'embedded-js',
+  'embedded-css',
+  'edit-template',
+  'edit-js',
+  'edit-css',
+];
+interface CardContext {
+  repository: string;
+  packageName: string;
+  cardId?: string;
+  modelId?: string;
+}
+
+export = declareInjections({
+  writers: 'hub:writers',
+  indexers: 'hub:indexers',
+  searchers: 'hub:searchers',
+  currentSchema: 'hub:current-schema',
+},
+
+class CardServices {
+  writers: todo;
+  indexers: todo;
+  searchers: todo;
+  currentSchema: todo;
+
+  cardContextFromId(id: string) {
+    let [repository, packageName, cardId, modelId] = id.split(cardIdDelim);
+
+    return {
+      repository,
+      packageName,
+      cardId,
+      modelId,
+    };
+  }
+
+  cardContextToId({ repository, packageName, cardId, modelId }: CardContext) {
+    return [repository, packageName, cardId, modelId].filter(i => i != null).join(cardIdDelim);
+  }
+
+  // TODO should we care about the session of the caller who is requesting to load the card?
+  async loadCard(card: SingleResourceDoc) {
```

Comment:
I want to try to get the public API of this service correct, even if the implementation needs hacks as we're in the middle of transitioning. And I don't think this method belongs in the public API.

`loadCard` should take the card's identity and it should find the card document itself via `hub:searchers`.

This implies changing where `loadCard` is being called from as well.